src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -802,6 +802,12 @@ internal override SyntaxNode AsInterfaceMember(SyntaxNode m)
                                  .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.SemicolonToken))
                                  .WithBody(null);
 
+                    case SyntaxKind.OperatorDeclaration:
+                        return ((OperatorDeclarationSyntax)member)
+                                 .WithModifiers(default)
```

Comment:
doesn't is need to be declared static at least?

also, are conversion ops allowed?  not sure what the rules are there.

Reply:
For the static modifier, yes and it seems it also requires the public one.

For the conversion, no they aren't allowed in interfaces (in general, you can't have an interface type to be to and from).

If these rules are important to be enforced for the generator, show should they be? I could put public + static on the modifier since it seems fine, but what about the conversion ones? Do we throw an exception because it can't be handled?

Reply:
well, you can filter out other modifiers, but keep the allowed ones.  or you can force adding those modifiers if they're literally always required.

Reply:
Ended up doing a bit of both: static is always required so I add it no matter.

But public can be implicit if omitted so this one, I allow it if it was present.

For abstract and virtual, one of the 2 is required because of how this type of member works in an interface so I also allow them to stay.

Doe this suitable?